src/extract.rs

```diff
@@ -95,21 +95,40 @@ where
     T: FromRequest,
     T::Future: 'static,
 {
-    type Config = T::Config;
     type Error = Error;
-    type Future = LocalBoxFuture<'static, Result<Option<T>, Error>>;
+    type Future = FromRequestOptFuture<T::Future>;
+    type Config = T::Config;
 
     #[inline]
     fn from_request(req: &HttpRequest, payload: &mut Payload) -> Self::Future {
-        T::from_request(req, payload)
-            .then(|r| match r {
-                Ok(v) => ok(Some(v)),
-                Err(e) => {
-                    log::debug!("Error for Option<T> extractor: {}", e.into());
-                    ok(None)
-                }
-            })
-            .boxed_local()
+        FromRequestOptFuture {
+            fut: T::from_request(req, payload),
+        }
+    }
+}
+
+#[pin_project::pin_project]
+pub struct FromRequestOptFuture<Fut> {
+    #[pin]
+    fut: Fut,
+}
+
+impl<Fut, T, E> Future for FromRequestOptFuture<Fut>
+where
+    Fut: Future<Output = Result<T, E>>,
+    E: Into<Error>,
+{
+    type Output = Result<Option<T>, Error>;
+
+    fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        match self.project().fut.poll(cx) {
```

Comment:
little clearer with ready! macro

Reply:
Trying to reduce the dependent on futures crate. At some point in the future it would be obsolete for actix-web crate.
We can already get rid of it if we bump MSRV to 1.48. 

I personally don't mind write some more code for matching polls. What do you think of the idea?

Reply:
ready macro is in futures-core